downloads/swan-lake-release-notes/swan-lake-alpha3/RELEASE_NOTE.md

```diff
@@ -2,41 +2,928 @@
+
+```ballerina
+import ballerina/io;
+
+function add(int v1, int v2) returns int => v1 + v2;
+
+function compare(int v1, int v2) returns boolean => v1 < v2;
+
+public function main() {
+    // A variable of type `function` can hold any function value.
+    function f = add;
+    io:println("Process (add, 1, 2): ", process(f, 1, 2)); // Prints `Process (add, 1, 2): 3`
+    io:println("Process (compare, 1, 2): ", process(compare, 1, 2)); // Prints `Process (compare, 1, 2): 0`
+}
+
+function process(function func, int v1, int v2) returns int {
+    // A function of type `function` cannot be called directly.
+    // A function value assigned to a `function`-typed variable
+    // can only be called after the type is narrowed to the relevant type.
+    if (func is function (int, int) returns int) {
+        return func(v1, v2);
+    }
+    return 0;
+}
+```
+
+##### Improved lang library functions
+
+###### New `xml:text()` function
+
+This function can be used to select all the items in a sequence that are of type `xml:Text`.
+
+```ballerina
+xml name = xml `<name>Dan<middleName>Gerhard</middleName><!-- This is a comment -->Brown</name>`;
+xml:Text nameText = (name/*).text();
+io:println(nameText); // "DanBrown"
+```
+
+##### Bug Fixes
+
+To view bug fixes, see the [GitHub milestone for Swan Lake Alpha3](https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%22Ballerina+Swan+Lake+-+Alpha3%22+label%3AType%2FBug+label%3ATeam%2FCompilerFE).
 
 #### Runtime
--
+
+##### New runtime Java API to create errors
+
+A new runtime Java API is introduced to create errors.
+```java
+BError createError(Module module, String errorTypeName, BString message, BError cause, Object details)
+```
+The `createDistinctError` API has been deprecated and should not be used to create distinct errors. The new `createError` API can be used instead.
+
+##### Configurable Variables
+
+Improved error messages are introduced for configurable variables by including more information and the relevant `Config.toml` line numbers.
+
+##### Bug Fixes
+
+To view bug fixes, see the [GitHub milestone for Swan Lake Alpha3](https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%22Ballerina+Swan+Lake+-+Alpha3%22+label%3AType%2FBug+label%3ATeam%2FjBallerina).
 
 #### Standard Library
-- 
+
+##### Log Package Updates
+
+###### Introduced additional log levels and log functions
+
+There are 4 log levels: `DEBUG`, `ERROR`, `INFO`, and `WARN` and their respective functions: `printDebug`, `printError`, `printInfo`, and `printWarn`. To set the global log level, place the entry given below in the `Config.toml` file:
+```toml
+[log] 
+level = "[LOG_LEVEL]"
+```
+
+Each module can also be assigned its own log level. To assign a log level to a module, provide the following entry in the `Config.toml` file:
+```toml
+[[log.modules]] 
+name = "[MODULE_NAME]" 
+level = "[LOG_LEVEL]"
+```
```

Comment:
@MadhukaHarith92 It's better if we specify the format in which we expect the module name.

Reply:
Updated in 6754883

Reply:
Fixed